lambda-runtime/src/AWS/Lambda/Runtime.hs

```diff
@@ -0,0 +1,47 @@
+module AWS.Lambda.Runtime
+  ( module Exports
+  , Client.InternalLambdaClientError (..)
+  , run
+  )
+where
+
+
+import AWS.Lambda.Env             as Exports
+import AWS.Prelude
+import Control.Monad.Except (runExceptT)
+
+import qualified AWS.Lambda.Client as Client
+import qualified Data.Aeson        as JSON
+
+run
+  :: forall e env m . (MonadCatch m, MonadIO m, Exception e)
```

Comment:
The idea behind having an `Exception e` constraint and `e` in `forall` is such that a client/user can tell `the lambda runtime` the kind of exceptions thrown in `lambda function handler` that they would want it to catch and send as an error event versus catching all exceptions.

If we remove it and we decide to catch any exception and then subsequently send them out as event errors then we will potentially catch `async IO exceptions` which should ideally fail the current hot lambda runtime instance immediately instead of being sent out as error events.

An alternative approach is to have `(JSON.Value -> RIO env (Either AppErrors JSON.Value))` or `(JSON.Value -> RIO env (Either Text JSON.Value))`
where `AppErrors` represents an error that a handler can pass to the lambda runtime for sending as an error event. But this is actually quite similar to the current approach.